processor/filterprocessor/README.md

```diff
@@ -317,6 +317,15 @@ The same relationship applies to metrics and datapoints.
 If all span events for a span are dropped, the span will be left intact.
 If all datapoints for a metric are dropped, the metric will also be dropped.
 
+The filter processor also allows configuring an optional field, `error_mode`, which will determine how the processor reacts to errors that occur while processing an OTTL condition.
+
+| error_mode            | description                                                                                                                |
+|-----------------------|----------------------------------------------------------------------------------------------------------------------------|
+| ignore                | The processor ignores errors returned by conditions and continues on to the next condition.  This is the recommended mode. |
```

Comment:
If this is the recommended mode, why is the default to propagate?

Reply:
Unfortunately to set `Ignore` by default [would be a breaking change for any component already using OTTL](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/18693#discussion_r1109122387).  The component would go from returning the OTTL errors to ignoring them, but we've already give the users of these components the ability to rely on those errors, so to suddenly remove them feels like a breaking change.  

It means a ton more work for me bc now I have to do a deprecation process eventually if I want to switch it, but it follows our Breaking Changes process.

Reply:
I see. Alternatively we could just say that propagating an error is the recommended mode :D

Reply:
I really don't think it should be tho.  For a majority of users I believe Ignore to be the better approach as 99% of errors are going to be things like using ParseJSON on an attribute that isn't present on the data or a string that isn't JSON or trying to do IsMatch in a condition when the attribute doesn't exist. The linked issue holds more details, but I do think Ignore is the best mode.  

Now we could always leave to Propagate as the default forever since it is more traditional way to handle errors.  That'd save me some time lol

Reply:
TBH, I do think the default should be propagate. I think ignoring errors should be an intentional step by the user, rather than having errors go unnoticed. I would imagine errors would occur more frequently at the initial stages of configurations.

Reply:
That being said, i also see why ignoring errors in a production environment to avoid dropping data makes more sense as a recommendation.